tests/checks/integration/test_nginx.py

```diff
@@ -31,27 +32,81 @@ def setUp(self):
                     'nginx_status_url': 'http://localhost:44441/nginx_status/',
                     'tags': ['second'],
                 },
+                {
+                    'nginx_status_url': 'https://localhost:44442/https_nginx_status/',
+                    'tags': ['ssl_enabled'],
+                    'ssl_validation': True,
+                },
+                {
+                    'nginx_status_url': 'https://localhost:44442/https_nginx_status/',
+                    'tags': ['ssl_disabled'],
+                    'ssl_validation': False,
+                },
             ]
         }
 
-    def test_nginx(self):
+    def test_nginx_one_connection(self):
         nginx = load_check('nginx', self.config, self.agent_config)
+
+        # Testing that connection will work with instance 0
         nginx.check(self.config['instances'][0])
+
+        # Checking that only one metric is of type 'nginx.net.connections'
         r = nginx.get_metrics()
         self.assertEquals(len([t for t in r if t[0] == "nginx.net.connections"]), 1, r)
 
+    def test_nginx_tags(self):
+        nginx = load_check('nginx', self.config, self.agent_config)
+
+        # Testing that connection will work with instance 1
         nginx.check(self.config['instances'][1])
+
+        # Checking that 'tags' attribute of some result is equal to 'tags' attribute in config for instance 1
         r = nginx.get_metrics()
         self.assertEquals(r[0][3].get('tags'), ['first_one'])
+
+        # Checking that each 'nginx.can_connect' service check's 'tags' attribute match expected host/port from config
         service_checks = nginx.get_service_checks()
         can_connect = [sc for sc in service_checks if sc['check'] == 'nginx.can_connect']
         for i in range(len(can_connect)):
             self.assertEquals(set(can_connect[i]['tags']), set(['host:localhost', 'port:44441']), service_checks)
 
+    def test_nginx_ssl_validation_enabled(self):
+        # Note: Throws an SSLError, because we're attempting to connect to an https endpoint with a self-signed
+        #       certificate. In addition, this throws an InsecurePlatformWarning. Both of these are expected;
+        #       versions of Python < 2.7.9 have restrictions in their ssl module limiting the configuration
+        #       urllib3 can apply. (https://urllib3.readthedocs.org/en/latest/security.html#insecurerequestwarning)
+        nginx = load_check('nginx', self.config, self.agent_config)
+
+        # Testing that connection will FAIL with instance 4
+        try:
+            nginx.check(self.config['instances'][4])
+        except Exception as e:
+            pass
+        self.assertEquals(type(e), requests.exceptions.SSLError)
```

Comment:
Wouldn't it be possible to use `assertRaises` to make it a little cleaner (to avoid having a pass and using `e` that isn't really supposed to be defined in this block ?